microsoft-edge/devtools-guide-chromium/evaluate-performance/unminify.md

```diff
@@ -0,0 +1,106 @@
+
+   In the following example, the source maps are served from the same server directory as the production code:
+
+   ![The source maps are served side-by-side with production code.](images/unminify-perf-profile-sourcemaps.msft.png)
+<!-- this was the best way I could come up with to describe the folder structure. https://docs.microsoft.com/en-us/dotnet/core/porting/project-structure is an example that went in a different direction but if you like it, please replace this image with a diagram -->
+
+   <!-- a text-based format that could replace the Sources > Page tab screenshot:
+   ```
+   hooks.ts
+   index.js
+   index.ts
+   isLoki
+   ```
+   -->
+
+   **Record a performance profile:**
+
+1. In the **Performance** tool, click the **Record** button, to record a performance profile for the page you want to analyze.  For details, see [Record runtime performance](/microsoft-edge/devtools-guide-chromium/evaluate-performance/reference).
+
+   <!-- If you are just quickly trying out the feature, you can record a performance profile while you scroll through a couple of emails and click them at your inbox at the Outlook web app. -->
+
+   ![Recording a performance profile.](../media/evaluate-performance-performance-record-highlight.msft.png)
+
+   A performance profile is recorded, in the **Performance** tool.
+
+1. In the **Performance** tool, click in the main area (the call stack), and use the mouse wheel to zoom in.  Click and drag left and right.  Look at the function names in the call stack.  Select a small section of the profile and look at the call stack.  Note the timestamp range.
+
+   The minified call stack in the **Performance** tool shows function names as **b**, **O**, **Xt**, **bn**, **n**, **i**, or **Yn**.
+
+   ![The minified call stack in the performance profile.](images/minified-perf-profile.msft.png)
+
+   **Unminify the code:**
+
+1. Click the **Unminify** button:
+   ![The Unminify button in the Performance tool.](images/perf-profile-unminify-button.msft.png)
+
+   In DevTools, a popup briefly appears, reading: **Status: creating unminfied file**.  Then the **Save As** dialog box opens.  The default filename is a pattern starting with date and time like `yymmddThhmmss`, such as: `Profile-20220307T164948-unminified.json`.  The default directory is the **Downloads** directory.
+
+1. Select a filename and directory and then click the **Save** button.
+
+   The **Performance** tool creates an unminified version of the recorded performance profile.  In the unminified version of the performance profile, some of the minified function names in the call stack are prepended by more meaningful function names from your source maps, where possible.  Some of the minified production function names might not be decorated with more meaningful, source-mapped names.
```

Comment:
```suggestion
   The **Performance** tool creates a new version of the recorded performance profile with some function names from the flame chart replaced by more meaningful function names using your source maps. Because source maps don't always provide the necessary information for the **Performance** tool to map all function names, some of the minified production names might still appear.
```

Reply:
If there is a simple way to explain why we can't map all function names, we should document it here.